### PR TITLE
Migrate our CI/CD from github actions to jenkins (CAS-397)

### PR DESCRIPTION
**Pull request by jkryl**
A race condition in nexus tests is fixed as well.

The first stage: we run jenkins along with github actions. When
it is proved to work we will remove github actions .github directory
completely.

### CODE REVIEW

File: doc/jenkins.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
+# Jenkins
+
+Jenkins is used to implement CI/CD pipeline for Mayastor. We make use of jenkins
+pipeline plugin that allows us to maintain CI/CD as a code that is part of
+Mayastor repository (Jenkinsfile). The goal of this document is to describe
+steps needed to set up the jenkins including the test executors (slaves in
+jenkins terminology). Details differ depending which environment the jenkins is
+set up in: AWS EC2 cloud, server with KVM guests, etc. We describe the most
+important common steps for all environments. We make use of NixOS capabilities
+for system configuration of nodes (as opposed to using ansible, salt, etc.).
```
Comment (paulyoong) [inline doc/jenkins.md:10]: Small nit - I think the 'j' in jenkins should be capitalised throughout ("Jenkins")

File: doc/jenkins.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+   imports = [ ./hardware.nix ];
+
+   services.openssh.enable = true;
+   services.jenkins.enable = true;
+
+   networking.firewall.enable = false;
+   networking.hostName = "ci-master";
+
+   # Skip this block if the system comes with preconfigured network (in aws).
+   # Replace the values below by something valid for your env.
```
Comment (paulyoong) [inline doc/jenkins.md:29]: Small nit - "...below **with** something"

**Review by paulyoong — APPROVED** [review]
:+1:

**Review by jonathan-teh — APPROVED** [review]

**Review by blaisedias — COMMENTED** [review]
LGTM

File: doc/jenkins.md (hunk 1 above)
Comment (jkryl) [reply doc/jenkins.md:10]: thanks. will do 👍

File: doc/jenkins.md (hunk 2 above)
Comment (jkryl) [reply doc/jenkins.md:29]: thanks 👍

**Pull request merged**